torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1069,6 +1069,34 @@ def sample_inputs_linalg_norm(op_info, device, dtype, requires_grad):
         return inputs
 
 
+def sample_inputs_batch_norm(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    # Ordered as: input shape, kwargs for training, momentum, eps
+    cases = (
+        ((S, S, S), {'training': False, 'momentum': 0.5, 'eps': 0.6}),
+        ((3, 2, 4), {'training': True, 'momentum': -1.2}),
+        ((3, 1), {'training': True, 'momentum': 0.0}),
+        ((2, 1), {'training': True, 'momentum': 0.2}),
+        ((1, 1, 2), {'training': False, 'momentum': 0.5}),
+        ((3, 2, 3, 4), {'training': True, 'momentum': -1.0, 'eps': 0.5}),
+        ((1, 2, 2), {})
+    )
+
+    def generator():
+        for input_shape, kwargs in cases:
+            # args: running mean, running var, weight and bias should necessarily be of shape: (C,)
+            channels = input_shape[1]
+            yield SampleInput(
+                make_arg(input_shape),
+                args = (
+                    make_arg(channels), make_arg(channels), make_arg(channels), make_arg(channels)
```

Comment:
For `running_mean`, don't set the `requires_grad` to `True`.

Something like this:
https://github.com/pytorch/pytorch/blob/741accb11e3eed6c2eb8171b200a8adadf1616fb/torch/testing/_internal/common_methods_invocations.py#L2470-L2471

Reply:
Thanks for the suggestion, @kshitij12345. This does help in getting rid of the error, and there are now real grad failures. Will take a look and see what's the issue. Thanks!

Reply:
Fixed now, `running_mean` and `running_var` needed to be of value > 0. Thanks, @kshitij12345 for the help!